tests/src/SplitInput.cpp

```diff
@@ -0,0 +1,109 @@
+#include "widgets/splits/SplitInput.hpp"
+
+#include "common/Literals.hpp"
+#include "controllers/accounts/AccountController.hpp"
+#include "controllers/commands/Command.hpp"
+#include "controllers/commands/CommandController.hpp"
+#include "controllers/hotkeys/HotkeyController.hpp"
+#include "mocks/EmptyApplication.hpp"
+#include "singletons/Emotes.hpp"
+#include "singletons/Fonts.hpp"
+#include "singletons/Paths.hpp"
+#include "singletons/Theme.hpp"
+#include "singletons/WindowManager.hpp"
+#include "widgets/Notebook.hpp"
+#include "widgets/splits/Split.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <QDebug>
+#include <QString>
+
+using namespace chatterino;
+using ::testing::Exactly;
+
+namespace {
+
+class MockApplication : mock::EmptyApplication
+{
+public:
+    MockApplication()
+        : windowManager(this->paths)
+    {
+    }
+    Theme *getThemes() override
+    {
+        return &this->theme;
+    }
+
+    HotkeyController *getHotkeys() override
+    {
+        return &this->hotkeys;
+    }
+
+    Fonts *getFonts() override
+    {
+        return &this->fonts;
+    }
+
+    WindowManager *getWindows() override
+    {
+        return &this->windowManager;
+    }
+
+    AccountController *getAccounts() override
+    {
+        return &this->accounts;
+    }
+
+    CommandController *getCommands() override
+    {
+        return &this->commands;
+    }
+
+    IEmotes *getEmotes() override
+    {
+        return &this->emotes;
+    }
+
+    Theme theme;
+    HotkeyController hotkeys;
+    Fonts fonts;
+    Paths paths;
+    WindowManager windowManager;
+    AccountController accounts;
+    CommandController commands;
+    Emotes emotes;
+};
+
+class SplitInputFixture : public ::testing::Test
+{
+protected:
+    SplitInputFixture()
+        : split(new Split(nullptr))
```

Comment:
This split is never deleted, is that intentional?

Reply:
Not intentional, no - but yolo since it's just in tests